### ISSUE TITLE
vcpu count plugin

### ISSUE THREAD

**Issue by Kennric** [osuosl/iam#48]

**Comment by ElijahCaine** [comment]
Questions to answer before writing the plugin:
[ ] What is the unit of measurement for a VCPU?
[ ] Which if our API's provide this information and in what form?

**Comment by matthewrsj** [comment]
I would like to work on this

**Comment by matthewrsj** [comment]
Looks like the only available fields regarding vcpus are the count of vcpus available:

```
...
"oper_vcpus"=>8,
...
"custom_beparams"=>{"minmem"=>2048,
                    "vcpus"=>8,
                    "maxmem"=>2048},
...
"beparams"=>{"auto_balance"=>true,
             "vcpus"=>8,
             ...
...
```

**Comment by ElijahCaine** [comment]
@matthewrsj In that case the unit of `vcpu` (or I guess `cpu` if we're also working with physical hardware?) is most appropriate.

**Comment by matthewrsj** [comment]
@ElijahCaine I see the light. Type changed to `vcpu`

**Comment by matthewrsj** [comment]
Resolved by #62
